apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.html.ts

```diff
@@ -0,0 +1,642 @@
+      </button>
+    </form>
+  `;
+}
+
+function uploadAssessmentGroupsModal(csrfToken: string) {
+  return html`
+    <div class="container-fluid">
+      <div
+        class="modal fade"
+        id="uploadAssessmentGroupsModal"
+        tabindex="-1"
+        role="dialog"
+        aria-labelledby="uploadAssessmentGroupsModalLabel"
+      >
+        <div class="modal-dialog" role="document">
+          <div class="modal-content">
+            <div class="modal-header">
+              <h1 class="h4 modal-title" id="uploadAssessmentGroupsModalLabel">
+                Upload new group assignments
+              </h1>
+            </div>
+            <div class="modal-body">
+              <p>Upload a CSV file in the format of:</p>
+            </div>
+            <div class="modal-body">
+              <code class="mb-0 text-dark">
+                groupName,UID<br />
+                groupA,one@example.com<br />
+                groupA,two@example.com<br />
+                groupB,three@example.com<br />
+                groupB,four@example.com</code
+              >
+              <!-- Closing code tag not on its own line to improve copy/paste formatting -->
+            </div>
+            <div class="modal-body">
+              <p>
+                The <code>groupname</code> column should be a unique identifier for each group. To
+                change the grouping, link uids to the groupname.
+              </p>
+              <form name="upload-assessment-group-form" method="POST" enctype="multipart/form-data">
+                <div class="form-group">
+                  <div class="custom-file">
+                    <input
+                      type="file"
+                      accept=".csv"
+                      name="file"
+                      class="custom-file-input"
+                      id="uploadAssessmentGroupsFileInput"
+                    />
+                    <label class="custom-file-label" for="uploadAssessmentGroupsFileInput"
+                      >Choose CSV file</label
+                    >
+                  </div>
+                </div>
+                <div class="d-flex justify-content-end">
+                  <div class="form-group mb-0">
+                    <input type="hidden" name="__action" value="upload_assessment_groups" />
+                    <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+                    <button type="button" class="btn btn-secondary" data-dismiss="modal">
+                      Cancel
+                    </button>
+                    <button type="submit" class="btn btn-primary">Upload</button>
+                  </div>
+                </div>
+              </form>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  `;
+}
+
+function autoAssessmentGroupsModal({
+  groupMin,
+  groupMax,
+  csrfToken,
+}: {
+  groupMin: number;
+  groupMax: number;
+  csrfToken: string;
+}) {
+  return html` <div class="container-fluid">
```

Comment:
Nit (pair this with a trailing newline after the closing `<div>` before the backtick, then run Prettier again):

```suggestion
  return html`
    <div class="container-fluid">
```

Reply:
This line has been removed after implementing the modal in [d845748](https://github.com/PrairieLearn/PrairieLearn/pull/9550/commits/d845748f0ca527176c6fd7e5ed5ec08d7abafcbf).